scanpipe/migrations/0023_migrate_dependencies.py

```diff
@@ -0,0 +1,52 @@
+# Generated by Django 4.0.6 on 2022-08-12 23:13
+
+from django.db import migrations
+from django.db.models import F, Q
+from django.db.models.functions import Concat
+
+
+def migrate_dependencies_to_discovereddependencies(apps, schema_editor):
+    DiscoveredPackage = apps.get_model('scanpipe', 'DiscoveredPackage')
+    DiscoveredDependency = apps.get_model('scanpipe', 'DiscoveredDependency')
+
+    package_with_dependencies = DiscoveredPackage.objects.exclude(dependencies_data=[])
+
+    for package in package_with_dependencies:
+        for dependency_data in package.dependencies_data:
+            project = package.project
+
+            # Remove non-supported fields from the data dict
+            dependency_data.pop("extra_data", None)
+            dependency_data.pop("resolved_package", None)
+
+            for_package = None
+            for_package_uid = dependency_data.get("for_package_uid")
+            p = project.discoveredpackages.filter(package_uid=for_package_uid)
+            if p.exists():
+                for_package = project.discoveredpackages.get(
+                    package_uid=for_package_uid
+                )
```

Comment:
This approach works but it triggers 2 queries.
Another possibility is to directly get your object and catch if it is not available:
```
try:
    for_package = project.discoveredpackages.get(package_uid=for_package_uid)
except (ObjectDoesNotExist, MultipleObjectsReturned):
    for_package = None
```